File: CS/FontDialogCommonForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Repository;

namespace CustomFontEditControl
{
    internal partial class FontDialogCommonForm : DevExpress.XtraEditors.XtraForm
    {
        RepositoryItemCustomFontEdit currentRepository;
        internal BaseEdit currentEditor { get; set; }

        internal FontDialogCommonForm(RepositoryItemCustomFontEdit Owner)
        {
            InitializeComponent();
            currentRepository = Owner;
        }

        private void FontDialogCommonForm_Load(object sender, EventArgs e)
        {
            CreateFontNames();
            CreateFontStyles();
            CreateFontSizes();
            InitializeControlsValue();
            CustomControlBinder.AssignControls(listBoxControlFontName, textEditFontName, true);
            CustomControlBinder.AssignControls(listBoxControlFontStyle, textEditFontStyle, true);
            CustomControlBinder.AssignControls(listBoxControlFontSize, textEditFontSize, false);
            textEditFontSize.EditValueChanged += new EventHandler(ChangeSampleLabelFont);
            checkEditStrikeOut.EditValueChanged += new EventHandler(ChangeSampleLabelFont);
            checkEditUnderLine.EditValueChanged += new EventHandler(ChangeSampleLabelFont);

            CustomControlBinder.TextValueChanged += new CustomControlBinder.CustomControlBinderrEventHandler(ChangeSampleLabelFont);
        }

        private void ChangeSampleLabelFont(object sender, EventArgs e)
        {
            if (textEditFontName.EditValue != null)
            {
                labelControlSample.Font = CreateNewFontInstance();
            }
        }

        protected void InitializeControlsValue()
        {
            if (currentEditor != null)
            {
                Font EditorFont = (currentEditor.EditValue as Font);
                if (EditorFont != null)
	            {
                    textEditFontName.EditValue = EditorFont.Name;
                    textEditFontSize.EditValue = EditorFont.Size;
                    switch (EditorFont.Style)
                    {
                        case FontStyle.Bold | FontStyle.Italic:
                            textEditFontStyle.EditValue = "Bold Italic";
                            break;
                        case FontStyle.Bold:
                            textEditFontStyle.EditValue = "Bold";
                            break;
                        case FontStyle.Italic:
                            textEditFontStyle.EditValue = "Italic";
                            break;
                        case FontStyle.Regular:
                            textEditFontStyle.EditValue = "Regular";
                            break;
                        default:
                            break;
                    }
                    switch (EditorFont.Style)
                    {
                        case FontStyle.Strikeout:
                            checkEditStrikeOut.EditValue = true;
                            break;
                        case FontStyle.Underline:
                            checkEditUnderLine.EditValue = true;
                            break;
                        default:
                            break;
                    }   
	            }            
            }        
        }

        protected void CreateFontNames()
        {
            listBoxControlFontName.Items.Clear();
            foreach (var item in currentRepository.Items)
            {
                listBoxControlFontName.Items.Add(item.ToString());
            }
        }

        protected void CreateFontStyles()
        {
            listBoxControlFontStyle.Items.Clear();
            listBoxControlFontStyle.Items.AddRange(new object[] { "Regular", "Italic", "Bold", "Bold Italic" });
        }

        protected void CreateFontSizes()
        {
            listBoxControlFontSize.Items.Clear();
            listBoxControlFontSize.Items.AddRange(new object[] { 8,9,10,11,12,14,16,18,20,22,24,26,28,36,48,72 });
        }


        private void listBoxControlFontName_DrawItem(object sender, ListBoxDrawItemEventArgs e)
        {
            RepositoryItemCustomFontEdit.DrawFontBoxItem(e, currentRepository, false, e.Item.ToString());
        }

        private void listBoxControlFontStyle_DrawItem(object sender, ListBoxDrawItemEventArgs e)
        {
            RepositoryItemCustomFontEdit.DrawFontBoxItem(e, currentRepository, true, listBoxControlFontName.SelectedItem.ToString());
        }

        private void listBoxControlFontName_SelectedIndexChanged(object sender, EventArgs e)
        {
            listBoxControlFontStyle.Invalidate();
        }

        protected Font CreateNewFontInstance()
        {            
            FontFamily family = new FontFamily(textEditFontName.EditValue.ToString());
            FontStyle currentStyle = FontStyle.Regular;
            if (textEditFontStyle.EditValue != null)
            {
                switch (textEditFontStyle.EditValue.ToString())
                {
                    case "Regular":
                        currentStyle = FontStyle.Regular;
                        break;
                    case "Italic":
                        currentStyle = FontStyle.Italic;
                        break;
                    case "Bold":
                        currentStyle = FontStyle.Bold;
                        break;
                    case "Bold Italic":
                        currentStyle = FontStyle.Bold | FontStyle.Italic;
                        break;
                    default:
                        break;
                }                    
            }
            if ((bool)checkEditUnderLine.EditValue) currentStyle = currentStyle | FontStyle.Underline;
            if ((bool)checkEditStrikeOut.EditValue) currentStyle = currentStyle | FontStyle.Strikeout;
            if (textEditFontSize.EditValue != null && Convert.ToInt32(textEditFontSize.EditValue.ToString()) > 0)
                return new Font(family, Convert.ToInt32(textEditFontSize.EditValue.ToString()), currentStyle);    
            else
                return new Font(family, 10, currentStyle);    
        }

        private void FontDialogCommonForm_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (this.DialogResult == DialogResult.OK)
            {
                currentEditor.EditValue = labelControlSample.Font;
            }
            textEditFontSize.EditValueChanged -= new EventHandler(ChangeSampleLabelFont);
            checkEditStrikeOut.EditValueChanged -= new EventHandler(ChangeSampleLabelFont);
            checkEditUnderLine.EditValueChanged -= new EventHandler(ChangeSampleLabelFont);
            CustomControlBinder.DisposeControlBinder();
        }
    }
}

File: VB/FontDialogCommonForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls

Namespace CustomFontEditControl

    Friend Partial Class FontDialogCommonForm
        Inherits XtraForm

        Private currentRepository As RepositoryItemCustomFontEdit

        Friend Property currentEditor As BaseEdit

        Friend Sub New(ByVal Owner As RepositoryItemCustomFontEdit)
            InitializeComponent()
            currentRepository = Owner
        End Sub

        Private Sub FontDialogCommonForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            CreateFontNames()
            CreateFontStyles()
            CreateFontSizes()
            InitializeControlsValue()
            CustomControlBinder.AssignControls(listBoxControlFontName, textEditFontName, True)
            CustomControlBinder.AssignControls(listBoxControlFontStyle, textEditFontStyle, True)
            CustomControlBinder.AssignControls(listBoxControlFontSize, textEditFontSize, False)
            AddHandler textEditFontSize.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            AddHandler checkEditStrikeOut.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            AddHandler checkEditUnderLine.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            AddHandler TextValueChanged, New CustomControlBinderrEventHandler(AddressOf ChangeSampleLabelFont)
        End Sub

        Private Sub ChangeSampleLabelFont(ByVal sender As Object, ByVal e As EventArgs)
            If textEditFontName.EditValue IsNot Nothing Then
                labelControlSample.Font = CreateNewFontInstance()
            End If
        End Sub

        Protected Sub InitializeControlsValue()
            If currentEditor IsNot Nothing Then
                Dim EditorFont As Font = TryCast(currentEditor.EditValue, Font)
                If EditorFont IsNot Nothing Then
                    textEditFontName.EditValue = EditorFont.Name
                    textEditFontSize.EditValue = EditorFont.Size
                    Select Case EditorFont.Style
                        Case FontStyle.Bold Or FontStyle.Italic
                            textEditFontStyle.EditValue = "Bold Italic"
                        Case FontStyle.Bold
                            textEditFontStyle.EditValue = "Bold"
                        Case FontStyle.Italic
                            textEditFontStyle.EditValue = "Italic"
                        Case FontStyle.Regular
                            textEditFontStyle.EditValue = "Regular"
                        Case Else
                    End Select

                    Select Case EditorFont.Style
                        Case FontStyle.Strikeout
                            checkEditStrikeOut.EditValue = True
                        Case FontStyle.Underline
                            checkEditUnderLine.EditValue = True
                        Case Else
                    End Select
                End If
            End If
        End Sub

        Protected Sub CreateFontNames()
            listBoxControlFontName.Items.Clear()
            For Each item In currentRepository.Items
                listBoxControlFontName.Items.Add(item.ToString())
            Next
        End Sub

        Protected Sub CreateFontStyles()
            listBoxControlFontStyle.Items.Clear()
            listBoxControlFontStyle.Items.AddRange(New Object() {"Regular", "Italic", "Bold", "Bold Italic"})
        End Sub

        Protected Sub CreateFontSizes()
            listBoxControlFontSize.Items.Clear()
            listBoxControlFontSize.Items.AddRange(New Object() {8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72})
        End Sub

        Private Sub listBoxControlFontName_DrawItem(ByVal sender As Object, ByVal e As ListBoxDrawItemEventArgs)
            RepositoryItemCustomFontEdit.DrawFontBoxItem(e, currentRepository, False, e.Item.ToString())
        End Sub

        Private Sub listBoxControlFontStyle_DrawItem(ByVal sender As Object, ByVal e As ListBoxDrawItemEventArgs)
            RepositoryItemCustomFontEdit.DrawFontBoxItem(e, currentRepository, True, listBoxControlFontName.SelectedItem.ToString())
        End Sub

        Private Sub listBoxControlFontName_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            listBoxControlFontStyle.Invalidate()
        End Sub

        Protected Function CreateNewFontInstance() As Font
            Dim family As FontFamily = New FontFamily(textEditFontName.EditValue.ToString())
            Dim currentStyle As FontStyle = FontStyle.Regular
            If textEditFontStyle.EditValue IsNot Nothing Then
                Select Case textEditFontStyle.EditValue.ToString()
                    Case "Regular"
                        currentStyle = FontStyle.Regular
                    Case "Italic"
                        currentStyle = FontStyle.Italic
                    Case "Bold"
                        currentStyle = FontStyle.Bold
                    Case "Bold Italic"
                        currentStyle = FontStyle.Bold Or FontStyle.Italic
                    Case Else
                End Select
            End If

            If CBool(checkEditUnderLine.EditValue) Then currentStyle = currentStyle Or FontStyle.Underline
            If CBool(checkEditStrikeOut.EditValue) Then currentStyle = currentStyle Or FontStyle.Strikeout
            If textEditFontSize.EditValue IsNot Nothing AndAlso Convert.ToInt32(textEditFontSize.EditValue.ToString()) > 0 Then
                Return New Font(family, Convert.ToInt32(textEditFontSize.EditValue.ToString()), currentStyle)
            Else
                Return New Font(family, 10, currentStyle)
            End If
        End Function

        Private Sub FontDialogCommonForm_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            If DialogResult = DialogResult.OK Then
                currentEditor.EditValue = labelControlSample.Font
            End If

            RemoveHandler textEditFontSize.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            RemoveHandler checkEditStrikeOut.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            RemoveHandler checkEditUnderLine.EditValueChanged, New EventHandler(AddressOf ChangeSampleLabelFont)
            DisposeControlBinder()
        End Sub
    End Class
End Namespace
